Directory.Build.targets

```diff
@@ -68,4 +70,95 @@
   <Import Project="eng\targets\FSharp.Common.targets"  Condition="'$(MSBuildProjectExtension)' == '.fsproj'" />
   <Import Project="eng\targets\Wix.Common.targets"  Condition="'$(MSBuildProjectExtension)' == '.wixproj'" />
   <Import Project="eng\targets\Npm.Common.targets"  Condition="'$(MSBuildProjectExtension)' == '.npmproj'" />
+
+  <ItemGroup>
+    <PackageReference Include="xunit.runner.console" Version="2.4.1" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
```

Comment:
@natemcmaster so this xunit package reference is what's causing the installers to fail on normal CI, is there a way to condition this to only get included when doing restore for the helix target so it doesn't mess up the normal builds?  This xunit console runner is only needed for the handful of net461 payloads

Reply:
Here are two ideas which might help

1. Move this reference to language-specific files, like `eng/targets/Csharp.Common.props`
2. Add a property, like `/p:TestWithHelix=true` to the required command line parameters for running helix so you can conditionally import this reference when `Condition=" '$(TestWithHelix)' == 'true'"`

Reply:
Cool went with the explicit parameter which works, should I also move it to another C# specific targets file as well?

Reply:
I can't seem to get things building with the xunit reference in Csharp.Common.props, the references in directory.build.targets can't resolve the path to the xunit files.  Does the `CollectXunitConsoleRunner` target also need to move?

```
2019-02-06T11:10:19.6145324Z D:\a\1\s\Directory.Build.targets(114,7): error MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated. The path is not of a legal form. [D:\a\1\s\src\SiteExtensions\Microsoft.Web.Xdt.Extensions\tests\Microsoft.Web.Xdt.Extensions.Tests.csproj]
2019-02-06T11:10:19.6147254Z D:\a\1\s\Directory.Build.targets(114,7): error MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated. The path is not of a legal form. [D:\a\1\s\src\Shared\test\Shared.Tests\Microsoft.AspNetCore.Shared.Tests.csproj]
2019-02-06T11:10:19.6147967Z D:\a\1\s\Directory.Build.targets(114,7): error MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated. The path is not of a legal form. [D:\a\1\s\src\Middleware\HealthChecks.EntityFrameworkCore\test\Microsoft.Extensions.Diagnostics.HealthChecks.EntityFrameworkCore.Tests.csproj]

```